packages/gatsby/cache-dir/static-entry.js

```diff
@@ -49,6 +49,16 @@ const getPage = path => pagesObjectMap.get(path)
 
 const createElement = React.createElement
 
+const sanitizeHeadComponents = components => {
```

Comment:
you could rewrite this function to something like

```js
const sanitizeHeadComponents = components => {
  const componentsAsArray = Array.isArray(components) ? components : [components];

  return componentsAsArray.filter(Boolean);
};
```
The filter Boolean removes all falsy (undefined, null, fasle) values. I don't think you can have nested arrays.

Reply:
As of right now in my particular example I can get the following, just using Helmet:

```js
[
  [
    {
      $$typeof: Symbol(react.element),
      type: "title",
      key: "",
      ref: null,
      props: [Object],
      _owner: null,
      _store: {}
    }
  ],
  [],
  [],
  [],
  [],
  [],
  null,
  {
    $$typeof: Symbol(react.element),
    type: "link",
    key: "gatsby-plugin-feed-0",
    ref: null,
    props: {
      rel: "alternate",
      type: "application/rss+xml",
      title: undefined,
      href: "/rss.xml"
    },
    _owner: null,
    _store: {}
  },
  // Other component definitions
];
```

They're all valid React Children, so it makes sense. I wanted to use this as a starting point for sanitizing the input.

Reply:
ok, I missed what you were trying to do. Thanks for clarifying.